inc/class-core-sitemaps-renderer.php

```diff
@@ -31,10 +31,24 @@ class Core_Sitemaps_Renderer {
 	 * Core_Sitemaps_Renderer constructor.
 	 */
 	public function __construct() {
-		$stylesheet_url         = $this->get_sitemap_stylesheet_url();
-		$stylesheet_index_url   = $this->get_sitemap_index_stylesheet_url();
-		$this->stylesheet       = '<?xml-stylesheet type="text/xsl" href="' . esc_url( $stylesheet_url ) . '" ?>';
-		$this->stylesheet_index = '<?xml-stylesheet type="text/xsl" href="' . esc_url( $stylesheet_index_url ) . '" ?>';
+		/**
+		 * Filter whether the XSLT stylesheet is used when the sitemap is viewed in a browser.
+		 *
+		 * @param bool $use_stylesheet True if the XSLT stylesheet should be used, false otherwise.
+		 */
+		if ( apply_filters( 'core_sitemaps_use_stylesheet', true ) ) {
```

Comment:
Instead of introducing new very specific filters for this, I'd suggest just checking the return value of `get_sitemap_stylesheet_url()` / `get_sitemap_index_stylesheet_url()`.

Then developers could use the existing `core_sitemaps_stylesheet_url` and `core_sitemaps_stylesheet_index_url` filters to disable stylesheets.

```php
add_filter( 'core_sitemaps_stylesheet_url', '__return_false' );
add_filter( 'core_sitemaps_stylesheet_index_url', '__return_false' );
```

Reply:
yeah, that works...new commit coming soon